Rules/AvoidGlobalFunctions.cs

```diff
@@ -0,0 +1,205 @@
+﻿using Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Globalization;
+using System.Management.Automation.Language;
+
+namespace Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules
+{
+    [Export(typeof(IScriptRule))]
+    public class AvoidGlobalFunctions : AstVisitor, IScriptRule
+    {
+        List<DiagnosticRecord> records;
+        string fileName;
+
+        public IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            if (ast == null) throw new ArgumentNullException(Strings.NullAstErrorMessage);
+
```

Comment:
The statement must be on its own line with proper indentation or just use braces for consistency, even for one line statement